lib/graphql/metrics/tracer.rb

```diff
@@ -24,19 +25,24 @@ def trace(key, data, &block)
         # NOTE: Not all tracing events are handled here, but those that are are handled in this case statement in
         # chronological order.
         case key
-        when GRAPHQL_GEM_LEX_KEY
-          return setup_tracing(&block)
         when GRAPHQL_GEM_EXECUTE_MULTIPLEX_KEY
-          return setup_tracing(&block)
+          return capture_multiplex_start_time(&block)
         when GRAPHQL_GEM_PARSING_KEY
           return capture_parsing_time(&block)
         when GRAPHQL_GEM_VALIDATION_KEY
           context = possible_context
           return capture_validation_time(context, &block)
-        when GRAPHQL_GEM_ANALYZE_KEY
+        when GRAPHQL_GEM_ANALYZE_MULTIPLEX_KEY
+          # Ensures that we reset potentially long-lived PreContext objects between multiplexs. We reset at this point
```

Comment:
Is it possible to have a test case for this?

Reply:
Absolutely; in fact I could introduce a regression test similar to https://github.com/Shopify/arrive-server/pull/10384

Reply:
Updated w/ regression test